src/platform/linux/init.c

```diff
@@ -0,0 +1,125 @@
+
+#include <stdio.h>
+#include <string.h>
+#include <dlfcn.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/wait.h>
+
+#include <list.h>
+#include <component.h>
+#include <config.h>
+#include <resources.h>
+
+//#define ENABLE_TRACE
+#include <trace.h>
+
+static xmlNode *config;
+static list_t component_registry;
+static list_t resource_registry;
+static component_t component;
+
+static int allocate_resources(list_t *item, unsigned size, void *arg)
+{
+    resource_t *res = (resource_t *)((*item)->content);
+    return allocate_resource(res);
+}
+
+static int find_resources(list_t *item, unsigned size, void *arg)
```

Comment:
Shouldn't this actually be called "setup_resource"?

Reply:
`setup_resource` is already declared by an include. The rationale is that `setup_resources` is the call to setup all resources which then calls `find_resources` which finds the matching resources by checking the label and type and then sets up the matched resource with `setup_resource`.